src/pip/_internal/index.py

```diff
@@ -174,10 +174,11 @@ def _handle_get_page_fail(link, reason, url, meth=None):
     meth("Could not fetch URL %s: %s - skipping", link, reason)
 
 
-def _get_html_page(link, session=None):
+def _get_html_page(link, session=None, unresponsive_hosts=None):
```

Comment:
I was following the existing style on session, and since it's a new argument I put it at the end to preserve positional args. I too don't understand the "=None" and then a custom line of code to check if it's not set. For consistency it would be cleaner to remove all of this, including for session, and just have them as explicit arguments.

Reply:
I have left them as is, but opened a new PR for resolving this as I think the `=None` for the session can be removed too.